File: gyswp-admin/src/entities/indicator.ts
export class Indicators {
    id: string;
    title: string;
    paraphrase: string;
    measuringWay: string;
    creationTime: Date;
    creatorEmpeeId: string;
    creatorEmpName: string;
    creatorDeptId: number;
    creatorDeptName: string;
    deptIds: string;
    deptNames: string;
    expectedValue: number;
    cycleTime: number;
    cycleTimeName: string;
    achieveType: number = 1;
    sourceDocId: string;
    sourceDocName: string;
    endTime: Date;
    getDepts(): any[] {
        let depts = [];
        if (!this.deptIds) {
            this.deptIds = '';
        }
        if (!this.deptNames) {
            this.deptNames = '';
        }
        let ids = this.deptIds.split(',');
        let names = this.deptNames.split(',');
        let i = 0;
        for (let id of ids) {
            if (id) {
                depts.push({ id: id, name: names[i] });
            }
            i++;
        }
        return depts;
    }

    constructor(data?: any) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }
    init(data?: any) {
        if (data) {
            this.id = data["id"];
            this.title = data["title"];
            this.paraphrase = data["paraphrase"];
            this.measuringWay = data["measuringWay"];
            this.creationTime = data["creationTime"];
            this.creatorEmpeeId = data["creatorEmpeeId"];
            this.creatorEmpName = data["creatorEmpName"];
            this.creatorDeptId = data["creatorDeptId"];
            this.creatorDeptName = data["creatorDeptName"];
            this.deptIds = data["deptIds"];
            this.deptNames = data["deptNames"];
            this.expectedValue = data["expectedValue"];
            this.cycleTime = data["cycleTime"];
            this.cycleTimeName = data["cycleTimeName"];
            this.achieveType = data["achieveType"];
            this.sourceDocId = data["sourceDocId"];
            this.sourceDocName = data["sourceDocName"];
            this.endTime = data["endTime"];
        }
    }
    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["id"] = this.id;
        data["title"] = this.title;
        data["paraphrase"] = this.paraphrase;
        data["measuringWay"] = this.measuringWay;
        data["creationTime"] = this.creationTime;
        data["creatorEmpeeId"] = this.creatorEmpeeId;
        data["creatorEmpName"] = this.creatorEmpName;
        data["creatorDeptId"] = this.creatorDeptId;
        data["creatorDeptName"] = this.creatorDeptName;
        data["deptIds"] = this.deptIds;
        data["deptNames"] = this.deptNames;
        data["expectedValue"] = this.expectedValue;
        data["cycleTime"] = this.cycleTime;
        data["achieveType"] = this.achieveType;
        data["sourceDocId"] = this.sourceDocId;
        data["endTime"] = this.endTime;
        return data;
    }
    static fromJS(data: any): Indicators {
        let result = new Indicators();
        result.init(data);
        return result;
    }
    static fromJSArray(dataArray: any[]): Indicators[] {
        let array = [];
        dataArray.forEach(result => {
            let item = new Indicators();
            item.init(result);
            array.push(item);
        });
        return array;
    }
    clone() {
        const json = this.toJSON();
        let result = new Indicators();
        result.init(json);
        return result;
    }
}

File: aspnet-core/aspnet-core/src/GYSWP.Core/Indicators/Indicator.cs
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using GYSWP.GYEnums;

namespace GYSWP.Indicators
{
    /// <summary>
    /// 检查指标
    /// </summary>
    [Table("Indicators")]
    public class Indicator : Entity<Guid>, IHasCreationTime
    {

        /// <summary>
        /// 检查指标
        /// </summary>
        [Required]
        [StringLength(200)]
        public virtual string Title { get; set; }

        /// <summary>
        /// 指标释义
        /// </summary>
        [Required]
        [StringLength(2000)]
        public virtual string Paraphrase { get; set; }

        /// <summary>
        /// 测量方式
        /// </summary>
        [Required]
        [StringLength(2000)]
        public virtual string MeasuringWay { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        [Required]
        public virtual DateTime CreationTime { get; set; }

        /// <summary>
        /// 考核者人
        /// </summary>
        [Required]
        [StringLength(200)]
        public virtual string CreatorEmpeeId { get; set; }

        /// <summary>
        /// 考核者快照
        /// </summary>
        [StringLength(50)]
        public virtual string CreatorEmpName { get; set; }

        /// <summary>
        /// 考核部门
        /// </summary>
        [Required]
        public virtual long CreatorDeptId { get; set; }

        /// <summary>
        /// 考核部门快照
        /// </summary>
        [StringLength(100)]
        public virtual string CreatorDeptName { get; set; }

        /// <summary>
        /// 被考核部门
        /// </summary>
        [Required]
        public virtual string DeptIds { get; set; }

        /// <summary>
        /// 被考核部门快照
        /// </summary>
        public virtual string DeptNames { get; set; }

        /// <summary>
        /// 预期值
        /// </summary>
        [Required]
        public virtual decimal ExpectedValue { get; set; }

        /// <summary>
        /// 周期
        /// </summary>
        [Required]
        public virtual CycleTime CycleTime { get; set; }

        /// <summary>
        /// 达成条件
        /// </summary>
        [Required]
        public virtual AchieveType AchieveType { get; set; }

        /// <summary>
        /// 来源标准Id
        /// </summary>
        [Required]
        public virtual Guid SourceDocId { get; set; }
        /// <summary>
        /// 截止时间 当天23:59:59
        /// </summary>
        //[Required]
        //public virtual DateTime EndTime { get; set; }

        /// <summary>
        /// 是否启用
        /// </summary>
        [Required]
        public virtual bool IsAction { get; set; }
}
}
